echopype/visualize/api.py

```diff
@@ -0,0 +1,123 @@
+from typing import Union, List
+
+import xarray as xr
+
+from echopype.visualize.plot import _plot_echogram, FacetGrid, QuadMesh, T
+from echopype.echodata import EchoData
+
+
+def create_echogram(
+    data: Union[EchoData, xr.Dataset],
+    frequency: Union[int, float, List[T], None] = None,
+    get_range: bool = False,
+    range_kwargs: dict = {},
+    water_level: Union[xr.DataArray, None] = None,
+    **kwargs,
+) -> Union[FacetGrid, QuadMesh]:
+    """Create an Echogram from an EchoData object or Sv and MVBS Dataset.
+
+    Parameters
+    ----------
+    data : EchoData or xr.Dataset
+        Echodata or Xarray Dataset to be plotted
+    frequency : int, float, or list of float or ints, optional
+        The frequency to be plotted.
+        Otherwise all frequency will be plotted.
+    get_range : bool
+        Flag as to whether range should be computed or not,
+        by default it will just plot range_bin as the yaxis.
+    range_kwargs : dict
+        Keyword arguments dictionary for computing range.
+        Keys are `env_params`, `waveform_mode`, and `encode_mode`.
+    water_level : xr.DataArray, optional
+        Water level data array for platform water level correction.
+    **kwargs: optional
+        Additional keyword arguments for xarray plot pcolormesh.
+
+    """
+    range_attrs = {
+        'long_name': 'Range',
+        'units': 'm',
+    }
+
+    if isinstance(frequency, list) and len(frequency) == 1:
+        frequency = frequency[0]
+
+    if isinstance(data, EchoData):
+        if data.sonar_model.lower() == 'ad2cp':
+            raise ValueError(
+                "Visualization for AD2CP sonar model is currently unsupported."
+            )
+        yaxis = 'range_bin'
+        variable = 'backscatter_r'
+        ds = data.beam
+        if get_range:
+            yaxis = 'range'
+
+            if data.sonar_model.lower() == 'azfp':
+                if 'azfp_cal_type' not in range_kwargs:
+                    range_kwargs['azfp_cal_type'] = 'Sv'
+                if 'env_params' not in range_kwargs:
+                    raise ValueError(
+                        "Please provide env_params in range_kwargs!"
+                    )
+            elif data.sonar_model.lower() == 'ek60':
+                if 'waveform_mode' not in range_kwargs:
+                    range_kwargs['waveform_mode'] = 'CW'
+
+                if 'encode_mode' not in range_kwargs:
+                    range_kwargs['encode_mode'] = 'power'
```

Comment:
EK60 should always be `waveform="CW"` and `encode_mode="power"` — anything else should raise an error